src/System.Collections/tests/Generic/Dictionary/Dictionary.Generic.Tests.netcoreapp.cs

```diff
@@ -220,5 +222,197 @@ public void EnsureCapacity_Generic_CapacityIsSetToPrimeNumberLargerOrEqualToRequ
+        {
+            Dictionary<TKey, TValue> dictionary = (Dictionary<TKey, TValue>)GenericIDictionaryFactory(count);
+            Assert.Equal(count, dictionary.Count);
+            // The smallest possible capacity size after clearing a dictionary is 3
+            dictionary.Clear();
+            dictionary.TrimExcess();
+            Assert.Equal(3, dictionary.EnsureCapacity(0));
+        }
+
+        [Theory]
+        [InlineData(85)]
+        [InlineData(89)]
+        public void TrimExcess_NoArguments_TrimsToAtLeastCount(int count)
+        {
+            var dictionary = new Dictionary<int, int>(20);
+            for (int i = 0; i < count; i++)
+            {
+                dictionary.Add(i, 0);
+            }
+            dictionary.TrimExcess();
+            Assert.InRange(dictionary.EnsureCapacity(0), count, int.MaxValue);
+        }
+
+        [Theory]
+        [InlineData(85)]
+        [InlineData(89)]
+        public void TrimExcess_WithArguments_OnDictionaryWithManyElementsRemoved_TrimsToAtLeastRequested(int finalCount)
+        {
+            const int InitToFinalRatio = 10;
+            int initialCount = InitToFinalRatio * finalCount;
+            var dictionary = new Dictionary<int, int>(initialCount);
+            Assert.InRange(dictionary.EnsureCapacity(0), initialCount, int.MaxValue);
+            for (int i = 0; i < initialCount; i++)
+            {
+                dictionary.Add(i, 0);
+            }
+            for (int i = 0; i < initialCount - finalCount; i++)
+            {
+                dictionary.Remove(i);
+            }
+            for (int i = InitToFinalRatio; i > 0; i--)
+            {
+                dictionary.TrimExcess(i * finalCount);
+                Assert.InRange(dictionary.EnsureCapacity(0), i * finalCount, int.MaxValue);
+            }
+        }
+        
+        [Theory]
+        [InlineData(1000, 900, 5000, 85, 89)]
+        [InlineData(1000, 400, 5000, 85, 89)]
+        [InlineData(1000, 900, 500, 85, 89)]
+        [InlineData(1000, 400, 500, 85, 89)]
+        [InlineData(1000, 400, 500, 1, 3)]
+        public void TrimExcess_NoArgument_TrimAfterEachBulkAddOrRemove_TrimsToAtLeastCount(int initialCount, int numRemove, int numAdd, int newCount, int newCapacity)
+        {
+            Random random = new Random(32);
+            var dictionary = new Dictionary<int, int>();
+            dictionary.TrimExcess();
+            Assert.InRange(dictionary.EnsureCapacity(0), dictionary.Count, int.MaxValue);
+
+            var initialKeys = new int[initialCount];
+            for (int i = 0; i < initialCount; i++)
+            {
+                initialKeys[i] = i;
+            }
+            random.Shuffle(initialKeys);
+            foreach (var key in initialKeys)
+            {
+                dictionary.Add(key, 0);
+            }
+            dictionary.TrimExcess();
+            Assert.InRange(dictionary.EnsureCapacity(0), dictionary.Count, int.MaxValue);
```

Comment:
Why a range? Shouldn't it always trim to the exact count?

Reply:
The reason it is a range is because EnsureCapacity returns capacity and capacity is always a prime number.
count is not necessarily a prime number but can be used as a lower bound in the assert range statement